language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/util/ServiceTemplateGenerator.java

```diff
@@ -89,7 +98,34 @@ public boolean initialized() {
     private ServiceTemplateGenerator(LanguageServerContext context) {
         context.put(SERVICE_TEMPLATE_GENERATOR_KEY, this);
         this.moduleListenerMetaDataMap = new ConcurrentHashMap<>();
-        CompletableFuture.runAsync(() -> initialize(context));
+        LSClientLogger clientLogger = LSClientLogger.getInstance(context);
+        String taskId = UUID.randomUUID().toString();
+        ExtendedLanguageClient languageClient = context.get(ExtendedLanguageClient.class);
+        CompletableFuture.runAsync(() -> {
+            clientLogger.logTrace("Loading listener symbols from the distribution");
+            if (languageClient != null) {
+                // Initialize progress notification
+                WorkDoneProgressCreateParams workDoneProgressCreateParams = new WorkDoneProgressCreateParams();
+                workDoneProgressCreateParams.setToken(taskId);
+                languageClient.createProgress(workDoneProgressCreateParams);
+
+                // Start progress
+                WorkDoneProgressBegin beginNotification = new WorkDoneProgressBegin();
+                beginNotification.setTitle(TITLE_INITIALIZE);
+                beginNotification.setCancellable(false);
+                beginNotification.setMessage("Loading listener symbols from the distribution");
```

Comment:
Final message shown will be `Initialize Service Template Generator: Loading listener symbols from the distribution` right? If yes, it's a bit lengthy. How about `Service Template Generator: Initializing...` where title is `Service Template Generator`. WDYT?

Reply:
+1

Reply:
Done